ygen/directory.go

```diff
@@ -114,6 +114,10 @@ func GoFieldNameMap(directory *Directory) map[string]string {
 // expected output (i.e., diffs don't appear simply due to reordering of the
 // Directory maps). If the names of the directories are not unique, which is
 // unexpected, an error is returned.
+// TODO(wenbli): this function's purpose is to check for name conflicts. This
```

Comment:
Is this implication here that we should remove this code after the IR refactor is complete (or at least the check that the length of the two maps is the same)?

Reply:
Yes that's right. I'll add to the TODO here to deprecate this API once the refactor is complete.

Reply:
SGTM.